lib/implementation.js

```diff
@@ -313,5 +324,58 @@ module.exports = function(self, options) {
     var stem = dotPath.split(/\./);
     return parseInt(stem[stem.length - 1]);
   };
+
+  // Invoked when the locale cannot be inferred from the session,
+  // hostname or URL prefix. The default implementation simply
+  // sets the locale to the default locale. You can override this
+  // method as you see fit.
+
+  self.guessLocale = function(req) {
+    req.locale = self.defaultLocale;
+  };
+  
+  self.enablecrossDomainSessionCache = function() {
+    self.crossDomainSessionCache = self.apos.caches.get('apostrophe-workflow-cross-domain-session-cache');
+  };
+  
+  self.acceptCrossDomainSessionToken = function(req) {
+    var crossDomainSessionToken = self.apos.launder.string(req.query.workflowCrossDomainSessionToken);
+    var sessionData;
+    
+    return async.series([
+      get,
+      clear
+    ], function(err) {
+
+      if (err) {
+        console.error(err);
+        return after();
+      }
+
+      _.each(_.keys(req.session), function(key) {
```

Comment:
Maybe it's that I'm not used to seeing this pattern much, but it seems like this may be more readable if the callback part of this code was factored out into a separate function. Thoughts?